### ISSUE TITLE
Virtuoso 37000 Error SP030 / Invalid character in SPARQL expression at '^'

### ISSUE THREAD

**Issue by mwjames** [openlink/virtuoso-opensource#220]
According to http://www.w3.org/TR/sparql11-query/#propertypaths, inverse paths should be possible but the following query just fails with: `Virtuoso 37000 Error SP030: SPARQL compiler, line 0: Invalid character in SPARQL expression at '^'` while the exact same query returns expected results on Jena Fuseki 1.0.2.

We can only run VOS 6.1 as it is the only package that available via `apt-get` and can be installed on the Travic-CI platform to run our [unit tests](https://travis-ci.org/SemanticMediaWiki/SemanticMediaWiki).

What is the problem and more importantly how to solve this? 

```
SPARQL query:
define sql:big-data-const 0 
#output-format:text/html
define sql:signal-void-variables 1 PREFIX wiki: <http://example.org/id/>
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX owl: <http://www.w3.org/2002/07/owl#>
PREFIX swivt: <http://semantic-mediawiki.org/swivt/1.0#>
PREFIX property: <http://example.org/id/Property-3A>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>
SELECT DISTINCT ?result WHERE {
?result swivt:wikiPageSortKey ?resultsk .
{
?v2 ^swivt:redirectsTo wiki:Window .
?result property:Has_target ?v2 .
} UNION {
?v4 ^swivt:redirectsTo wiki:Door .
?result property:Has_some_other_target ?v4 .
}
}
ORDER BY ASC(?resultsk) 
OFFSET 0

LIMIT 51
```

**Comment by HughWilliams** [comment]
What is the exact version of Virtuoso being used, run the command virtuoso-t -? which will return the minor 6.1.X version  ?

If you are installing a prebuilt binary from a Linux vendor then it is probably an older v6 version before property paths were supported, as the latest v6 & v7 versions do support this:

http://virtuoso.openlinksw.com/dataspace/doc/dav/wiki/Main/VirtTipsAndTricksSPARQL11PropertyPaths
In which case you would have to manually compile from the Virtuoso open source git repo at:

http://virtuoso.openlinksw.com/dataspace/doc/dav/wiki/Main/VOSGitUsage

**Comment by mwjames** [comment]
> What is the exact version of Virtuoso being used, run the command virtuoso-t -? which will return the minor 6.1.X version ?

6.1.6

> In which case you would have to manually compile from the Virtuoso open source git repo at:

Well, the thing is that we can only run our unit tests against an official available distribution to ensure that an user who installs SMW and Virtuoso is using an environment it is tested for but unfortunately Virtuoso 7 is not available via `apt-get` which means we can't test it.

I'd be happy to set-up a test run against Virtuoso 7.1 once it is available via `apt-get` but building a custom version based on a git repo can't be used to produce an indicator of stability (where unit tests reliably pass or fail) therefore as long as the `ap-get` version is outdated and not maintained, we probably have to advise our users that seek full SPARQL 1.1 support to use Jena Fuseki instead.

PS: It actually took us a while to get Virtuoso [as service](https://github.com/SemanticMediaWiki/SemanticMediaWiki/blob/master/build/travis/install-services.sh#L52) run on Travis-CI therefore I hope the next version doesn't change much in this respect in order for us to run automated unit tests.

**Comment by HughWilliams** [comment]
Hi Can you confirm if this problem is still an issue for you particularly with the latest Virtuoso 7.2 release ?

**Comment by mwjames** [comment]
> Hi Can you confirm if this problem is still an issue for you particularly with the latest Virtuoso 7.2 release ?

We are  able to run our unit tests on Travis [0] against a supported Debian package which in case of Virtuoso is only available as 6.1 and therefore we are unable to make any comments about Virtuoso 7.2. If at some point Virtuoso 7.2 is officially available we will be happy to run our tests against it.

```
 * Starting Virtuoso OpenSource Edition 6.1  virtuoso-opensource-6.1       [80G 
[74G[ OK ]
Warning: The current Virtuoso database uses default passwords.
The Virtuoso daemon has been disabled for security reasons.
You can reconfigure the package to change these passwords.
Setting up virtuoso-opensource (6.1.4+dfsg1-0ubuntu1) ...
Setting up virtuoso-vad-conductor (6.1.4+dfsg1-0ubuntu1) ...
Setting up virtuoso-vsp-startpage (6.1.4+dfsg1-0ubuntu1) ...
Setting up virtuoso-server (6.1.4+dfsg1-0ubuntu1) ...
+echo RUN=yes
+sudo tee -a /etc/default/virtuoso-opensource-6.1
RUN=yes
+sudo service virtuoso-opensource-6.1 start
 * Starting Virtuoso OpenSource Edition 6.1  virtuoso-opensource-6.1       [80G 
[74G[ OK ]
+isql-vt 1111 dba dba /home/travis/build/SemanticMediaWiki/SemanticMediaWiki/build/travis/virtuoso-sparql-permission.sql
Connected to OpenLink Virtuoso
Driver: 06.01.3127 OpenLink Virtuoso ODBC Driver
```

[0] https://s3.amazonaws.com/archive.travis-ci.org/jobs/53581877/log.txt

**Comment by mwjames** [comment]
We are still not able to access an official Virtuoso 7.2 dep packages therefore closing this issue.
